sdk/core/System.ClientModel/tests/gen.unit/ContextGeneratorTests.cs

```diff
@@ -1086,5 +1086,55 @@ public readonly struct Type {}
             Assert.AreEqual("JsonModel", result.GenerationSpec.TypeBuilders[0].Type.Name);
             Assert.AreEqual("TestProject", result.GenerationSpec.TypeBuilders[0].Type.Namespace);
         }
+
+        [Test]
+        public void BuilderForExperimentalPersistableHasSuppression()
+        {
+            string source =
+                """
+                  using System;
+                  using System.ClientModel.Primitives;
+                  using System.Diagnostics.CodeAnalysis;
+                  using System.Text.Json;
+
+                  namespace TestProject
+                  {
+                      [ModelReaderWriterBuildable(typeof(JsonModel))]
+                      [ModelReaderWriterBuildable(typeof(OtherModel))]
+                      public partial class LocalContext : ModelReaderWriterContext { }
+
+                      [Experimental("TEST001")]
+                      public class JsonModel : IJsonModel<JsonModel>
+                      {
+                          JsonModel IJsonModel<JsonModel>.Create(ref System.Text.Json.Utf8JsonReader reader, ModelReaderWriterOptions options) => new JsonModel();
+                          JsonModel IPersistableModel<JsonModel>.Create(BinaryData data, ModelReaderWriterOptions options) => new JsonModel();
+                          string IPersistableModel<JsonModel>.GetFormatFromOptions(ModelReaderWriterOptions options) => "J";
+                          void IJsonModel<JsonModel>.Write(System.Text.Json.Utf8JsonWriter writer, ModelReaderWriterOptions options) { }
+                          BinaryData IPersistableModel<JsonModel>.Write(ModelReaderWriterOptions options) => BinaryData.Empty;
+                      }
+
+                      [Experimental("TEST002")]
+                      public class OtherModel : IJsonModel<OtherModel>
+                      {
+                          OtherModel IJsonModel<OtherModel>.Create(ref System.Text.Json.Utf8JsonReader reader, ModelReaderWriterOptions options) => new OtherModel();
+                          OtherModel IPersistableModel<OtherModel>.Create(BinaryData data, ModelReaderWriterOptions options) => new OtherModel();
+                          string IPersistableModel<OtherModel>.GetFormatFromOptions(ModelReaderWriterOptions options) => "J";
+                          void IJsonModel<OtherModel>.Write(System.Text.Json.Utf8JsonWriter writer, ModelReaderWriterOptions options) { }
+                          BinaryData IPersistableModel<OtherModel>.Write(ModelReaderWriterOptions options) => BinaryData.Empty;
+                      }
+                  }
+                  """;
+
+            Compilation compilation = CompilationHelper.CreateCompilation(source);
+            var result = CompilationHelper.RunSourceGenerator(compilation, out var newCompilation, out var generatedSources);
+
```

Comment:
Asser the result has 2 buildables and the experimental id for both is the one we would expect.

You will also need tests where this experimental comes from a dependency.

You should consider other variations like collections of experimentals, obsolete and experimental, mix of experimentals and non.